### ISSUE TITLE
StackLayout Exception - regression on Android

### ISSUE THREAD

**Issue by attrib75** [xamarin/Xamarin.Forms#8349]
### Description
Exception when using stacklayout and/or listview
### Steps to Reproduce

1.  Create a page with a stacklayout and/or a listview.  Not sure which is the problem.  I can't tell from the stack trace
2.  Use the app as normal - intermittent issue in new XF

### Expected Behavior
Things work without crashing
### Actual Behavior
Crashes sometimes with the following stack trace:

11-01 11:50:21.983 I/MonoDroid(18734): UNHANDLED EXCEPTION:
11-01 11:50:21.988 I/MonoDroid(18734): System.NullReferenceException: Object reference not set to an instance of an object.
11-01 11:50:21.988 I/MonoDroid(18734):   at Xamarin.Forms.StackLayout.AlignOffAxis (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double widthConstraint, System.Double heightConstraint) [0x0004a] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:98 
11-01 11:50:21.989 I/MonoDroid(18734):   at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00034] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:53 
11-01 11:50:21.989 I/MonoDroid(18734):   at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x00158] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:266 
11-01 11:50:21.989 I/MonoDroid(18734):   at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:224 
11-01 11:50:21.989 I/MonoDroid(18734):   at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:784 
11-01 11:50:21.989 I/MonoDroid(18734):   at Xamarin.Forms.Layout.ResolveLayoutChanges () [0x0005c] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:392 
11-01 11:50:21.989 I/MonoDroid(18734):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <11a340ccc8de43f09c97400139266ef5>:0 
11-01 11:50:21.989 I/MonoDroid(18734):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <11a340ccc8de43f09c97400139266ef5>:0 
11-01 11:50:21.989 I/MonoDroid(18734):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr)
11-01 11:50:22.051 E/AppCenterCrashes(18734): Unhandled Exception from source=AndroidEnvironment
11-01 11:50:22.051 E/AppCenterCrashes(18734): System.NullReferenceException: Object reference not set to an instance of an object.
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Xamarin.Forms.StackLayout.AlignOffAxis (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double widthConstraint, System.Double heightConstraint) [0x0004a] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:98 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00034] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:53 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x00158] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:266 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:224 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:784 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Xamarin.Forms.Layout.ResolveLayoutChanges () [0x0005c] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:392 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <11a340ccc8de43f09c97400139266ef5>:0 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <11a340ccc8de43f09c97400139266ef5>:0 
11-01 11:50:22.051 E/AppCenterCrashes(18734):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.33(intptr,intptr)
11-01 11:50:22.278 D/Mono    (18734): Loading reference 5 of /storage/emulated/0/Android/data/com.beincarib/files/.__override__/Microsoft.AppCenter.Crashes.dll asmctx DEFAULT, looking for System.Core, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
11-01 11:50:22.278 D/Mono    (18734): Assembly Ref addref Microsoft.AppCenter.Crashes[0x6f6dfe1600] -> System.Core[0x6f5ffb1400]: 17
11-01 11:50:22.397 W/com.beincarib(18734): JNI RegisterNativeMethods: attempt to register 0 native methods for android.runtime.JavaProxyThrowable
11-01 11:50:22.398 D/Mono    (18734): DllImport searching in: '__Internal' ('(null)').
11-01 11:50:22.398 D/Mono    (18734): Searching for 'java_interop_jnienv_throw'.
11-01 11:50:22.398 D/Mono    (18734): Probing 'java_interop_jnienv_throw'.
11-01 11:50:22.398 D/Mono    (18734): Found as 'java_interop_jnienv_throw'.

### Basic Information

- Version with issue:  Xamarin 16.3.0277
- Last known good version:  any version before
- IDE: Visual Studio 16.3.7
- Platform Target Frameworks: 
  - iOS:   none
  - Android:  10.0.3.0
  - UWP:  None
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### Reproduction Link

I can't reproduce this except in my app as I don't know how to, I don't know what's causing it.   I can't share my app publicly

**Comment by hartez** [comment]
@attrib75 Can you give us some idea of what your ListView's ItemTemplate looks like? Are you using grouping? Is there a Header/Footer? Does the issue occur when you navigate around the app (i.e., pushing/popping pages)? Does it happen when you are scrolling around the ListView?

**Comment by hartez** [comment]
Also, you mention it's a regression on Android - what was the last working version of Xamarin.Forms (not Visual Studio, but the Forms NuGet package)? And which version did you update to when it broke?

**Comment by bpater-tp** [comment]
I have similar situation but in our project it crashes on iOS whereas it works fine on android.
`2019-11-03 15:34:52.369231+0100 Docs.iOS[20410:1062233] [AppCenterCrashes] ERROR: +[MSWrapperLogger MSWrapperLog:tag:level:]/10 Unhandled Exception:
System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.StackLayout.AlignOffAxis (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double widthConstraint, System.Double heightConstraint) [0x0004a] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:98 
  at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00034] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:53 
  at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x00158] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:266 
  at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:224 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:784 
  at Xamarin.Forms.Layout.ResolveLayoutChanges () [0x0005c] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:392 
  at Foundation.NSAsyncActionDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/Foundation/NSAction.cs:152 
--- End of stack trace from previous location where exception was thrown ---

  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:86 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:65 
  at Docs.iOS.Application.Main (System.String[] args) [0x00001] in /Users/noster/dev/tp/docsapp/Docs/Docs.iOS/Main.cs:13
2019-11-03 15:34:52.415884+0100 Docs.iOS[20410:1062233] Unhandled managed exception: Object reference not set to an instance of an object (System.NullReferenceException)
  at Xamarin.Forms.StackLayout.AlignOffAxis (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double widthConstraint, System.Double heightConstraint) [0x0004a] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:98 
  at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00034] in D:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:53 
  at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x00158] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:266 
  at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:224 
  at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:784 
  at Xamarin.Forms.Layout.ResolveLayoutChanges () [0x0005c] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:392 
  at Foundation.NSAsyncActionDispatcher.Apply () [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/Foundation/NSAction.cs:152 
--- End of stack trace from previous location where exception was thrown ---

  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:86 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/13.4.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:65 
  at Docs.iOS.Application.Main (System.String[] args) [0x00001] in /Users/noster/dev/tp/docsapp/Docs/Docs.iOS/Main.cs:13
`

**Comment by attrib75** [comment]
@bpater-tp it must be in iOS too.

**Comment by attrib75** [comment]
@hartez 

Can you give us some idea of what your ListView's ItemTemplate looks like? 
I'm using a syncfusion sflistview with a pretty complex template

Are you using grouping? 
no
Is there a Header/Footer? 
yes
Does the issue occur when you navigate around the app (i.e., pushing/popping pages)? Does it happen when you are scrolling around the ListView?
navigating to the page, back, then to the page again

Also, you mention it's a regression on Android - what was the last working version of Xamarin.Forms (not Visual Studio, but the Forms NuGet package)? 
version before the latest version
And which version did you update to when it broke?
latest version

**Comment by hartez** [comment]
> version before the latest version

I'm going to need a version number - "latest" version changes every time we release a new one :) 
Also, it's helpful to know if by "latest version" you mean "latest stable version", "latest pre-release version", or "latest nightly version".

**Comment by attrib75** [comment]
Yeah, I didn't think I'd get away with that :).  I am off this project now, but I'll see if I can dig back into it and get you that info.

**Comment by samhouts** [comment]
@attrib75 Thanks, we'll keep looking for it.

**Comment by samhouts** [comment]
@attrib75  Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!

**Comment by bpater-tp** [comment]
IIRC it was fixed by one of the 4.3 releases. thanks :)

**Comment by attrib75** [comment]
Thank you all!

**Comment by attrib75** [comment]
Update, I actually had this happen again in XF 4.4.0

**Comment by BillyMartin1964** [comment]
I got this from Google for a Pre-launch report. Seems similar, if not the same thing:

**Device(s) with issues
Google Pixel 2 (walleye) - Android 8.1
Google Pixel 2 (walleye) - Android 9
Samsung Galaxy S III (m0) - Android 4.3
Exceptions
at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.36(intptr,intptr)
at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.36(intptr,intptr)
java.lang.NoSuchMethodError: no method with name='isAttachedToWindow' signature='()Z' in class Landroid/view/ViewGroup;**

**Comment by AndreasLichtsinn** [comment]
I had the same issue with StackLayout.AlignOffAxis, in another discussion someone suggested that it could be connected with setting a binding on IsVisible property.
I have the following layout:

<AbsoluteLayout>
     <Image/>
     <StackLayout AbsoluteLayout.LayoutBounds="0,0,1,1" AbsoluteLayout.LayoutFlags="All">
     </StackLayout>
     <StackLayout AbsoluteLayout.LayoutBounds="0.5,0.5,-1,-1" 
            AbsoluteLayout.LayoutFlags="PositionProportional">
          <ActivityIndicator
             IsRunning="{Binding IsBusy}"
             IsVisible="{Binding IsVisible}" 
          />
    </StackLayout>
</AbsoluteLayout>

So removing the IsVisible Binding from the ActivityIndicator fixed it for me.

**Comment by theakkadian** [comment]
I have Xamarin.Forms 4.4.0+ and same issue is happening for a StackLayout that has 2 labels.  I had the ActivityIndicator somewhere in the page, and I tried to remove it, but that didn't help either.
This happens if I assign values to them in code, and in the second load of the page (launched using the Shell FlyoutItem / ShellContent / ContentTemplate method) it crashes with the error.  

If I only assign the first label inside of code, the app doesnt crash.  If assign back in code, then the app crashes.  
```
<StackLayout Spacing="0" Grid.Row="9" Grid.Column="0" Padding="10" HorizontalOptions="EndAndExpand" VerticalOptions="CenterAndExpand">
<Label x:Name="LeftMobileSensors" Style="{StaticResource LeftInfo}"  Text="."/>
<Label x:Name="LeftMobileSensorsDiscription" Style="{StaticResource LeftDescriptionInfo}"  Text="."/>
</StackLayout>
```

**Comment by theakkadian** [comment]
Replying on my own post.  I have found a workaround to avoid the crash, which is **removing** the "**Padding**" attribute from the StackLayout.  I am not sure why this is crashing, but on the second load of the page it causes the "Object not set to an instance .." issue.

**Comment by sethom** [comment]
This was happening to me too. It was happening when after I scroll down in a ListView and then scroll back up. Or when I clicked on a ListViewItem, navigate away and back again. So basically it was happening after the ListViewItem came back into view a second time. CachingStrategy="RetainElement".  XF version: 4.4.0.991640

I narrowed things down by removing all StackLayouts one at a time until I realized it was the ones with a Padding value assigned causing the issue. Once I removed Padding from StackLayouts, the problem went away (I am using padding on Frames and Grids with no problem within the same DataTemplate.)

Hope this helps!

**Comment by SGDAT** [comment]
Yep, I second @sethom. That was my issue too (thanks @sethom & @AndreasLichtsinn).

If I change the "IsVisible" property in the page at all (using a reference to the controls such as buttons and stack layouts) after the page has appeared, i get an exception if there is any padding on any of my stack layouts. I removed all padding that I was able to find and the exception went away.

I do have some syncfusion components around but not using any on the page that has the issue.

exception is :-

`System.NullReferenceException: Object reference not set to an instance of an object.
03-14 15:01:27.317 I/MonoDroid(10421):   at Xamarin.Forms.StackLayout.AlignOffAxis (Xamarin.Forms.StackLayout+LayoutInformation layout, Xamarin.Forms.StackOrientation orientation, System.Double widthConstraint, System.Double heightConstraint) [0x0004a] in d:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:98 
03-14 15:01:27.317 I/MonoDroid(10421):   at Xamarin.Forms.StackLayout.LayoutChildren (System.Double x, System.Double y, System.Double width, System.Double height) [0x00034] in d:\a\1\s\Xamarin.Forms.Core\StackLayout.cs:53 
03-14 15:01:27.317 I/MonoDroid(10421):   at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x00158] in d:\a\1\s\Xamarin.Forms.Core\Layout.cs:266 
03-14 15:01:27.317 I/MonoDroid(10421):   at Xamarin.Forms.Layout.OnSizeAllocated (System.Double width, System.Double height) [0x0000f] in d:\a\1\s\Xamarin.Forms.Core\Layout.cs:224 
03-14 15:01:27.317 I/MonoDroid(10421):   at Xamarin.Forms.VisualElement.SizeAllocated (System.Double width, System.Double height) [0x00000] in d:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:792 
03-14 15:01:27.317 I/MonoDroid(10421):   at Xamarin.Forms.Layout.ResolveLayoutChanges () [0x0005c] in d:\a\1\s\Xamarin.Forms.Core\Layout.cs:392 
03-14 15:01:27.317 I/MonoDroid(10421):   at Java.Lang.Thread+RunnableImplementor.Run () [0x00008] in <4a189ea3b82b48a089ac9002b2abc206>:0 
03-14 15:01:27.317 I/MonoDroid(10421):   at Java.Lang.IRunnableInvoker.n_Run (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <4a189ea3b82b48a089ac9002b2abc206>:0 
03-14 15:01:27.317 I/MonoDroid(10421):   at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.104(intptr,intptr)`

Xamarin Forms version is 4.5.0.356. Tested on android
